src/query/sql/src/planner/binder/bind_mutation/mutation_expression.rs

```diff
@@ -321,6 +341,12 @@ impl MutationExpression {
                     let mut rewriter = SubqueryDecorrelatorOptimizer::new(opt_ctx, None);
                     let s_expr = rewriter.optimize_sync(&s_expr)?;
 
+                    if !is_lazy_table && mutation_type != MutationType::Delete {
```

Comment:
Consider extracting this condition for better readability:

```rust
fn needs_all_columns(is_lazy_table: bool, mutation_type: &MutationType) -> bool {
    // Delete only needs row_id, Update/Merge need additional columns
    !is_lazy_table && *mutation_type != MutationType::Delete
}

if needs_all_columns(is_lazy_table, &mutation_type) {
    for column_index in bind_context.column_set().iter() {
        required_columns.insert(*column_index);
    }
}
```

Reply:
To ensure logical continuity, it's not suitable to extract a function. I've added comments.